lib/charms/bind/v0/dns_record.py

```diff
@@ -0,0 +1,696 @@
+# Copyright 2024 Canonical Ltd.
+# Licensed under the Apache2.0. See LICENSE file in charm source for details.
+
+"""Library to manage the integration with the Bind charm.
+
+This library contains the Requires and Provides classes for handling the integration
+between an application and a charm providing the `dns_record` integration.
+
+### Requirer Charm
+
+```python
+
+from charms.bind.v0.dns_record import DNSRecordRequires
+
+class DNSRecordRequirerCharm(ops.CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.dns_record = DNSRecordRequires(self)
+        self.framework.observe(self.dns_record.on.dns_record_data_available, self._handler)
+        ...
+
+    def _handler(self, events: DNSRecordDataAvailableEvent) -> None:
+        ...
+
+```
```

Comment:
This doesn't really show how to create a particular DNS record. Could that be included?

Reply:
How about now?